pkg/ingest/ingest.go

```diff
@@ -0,0 +1,20 @@
+package ingest
+
+import (
+	"github.com/thanos-community/obslytics/pkg/dataframe"
+	"github.com/thanos-community/obslytics/pkg/input"
+)
+
+type Ingestor interface {
+	// Ingest a single series. It expects the series' chunks to be sorted
+	// by timestamp.
+	Ingest(input.Series) error
+
+	// Indicate no more Ingest calls to be done. It allows finalizing the active
+	// data even when not reaching next sample time.
```

Comment:
even when not reaching next sample time? Can we elaborate?

Reply:
When ingestor is processing the data continuously, it can't turn them into the dataframe until the current sample ends.

Let's consider having a resolution of 1 day: it's still desirable to have the data from the today dumped as well, even when we haven't reached to the end of that sample yet.

At the end of the processing, we need to give the investor a chance to dump the last working set into the dataframe as well as last set of rows.

Reply:
Makes sense, ok, I think it's worth to put what you said as a comment. Also let's start comments with method name for goDoc to work properly. `Finalize allows to indicate no more ...`